_posts/2020-06-30-sparklyr-1.3.0-released/sparklyr-1.3.0.Rmd

```diff
@@ -0,0 +1,202 @@
+  4     4     3 ""
+  5   NaN     2 "d"
+
+```
+
+## Custom Serialization
+
+In addition to commonly used data serialization formats such as CSV, JSON, Parquet, and Avro, starting from sparklyr 1.3, customized data frame serialization and deserialization procedures implemented in R can also be run on Spark workers via the newly implemented `spark_read` and `spark_write` methods. We can see both of them in action through a quick example below, where `saveRDS` is called from a user-defined writer function to save all rows within a Spark data frame into 2 RDS files on disk, and `readRDS` is called from a user-defined reader function to read the data from the RDS files back to Spark:
+
+```{r echo=TRUE, eval=FALSE}
+library(sparklyr)
+
+sc <- spark_connect(master = "local")
+sdf <- sdf_len(sc, 7)
+paths <- c("/tmp/file1.RDS", "/tmp/file2.RDS")
+
+spark_write(sdf, writer = function(df, path) saveRDS(df, path), paths = paths)
+spark_read(sc, paths, reader = function(path) readRDS(path), columns = c(id = "integer"))
+```
+
+```
+# Source: spark<?> [?? x 1]
+     id
+  <int>
+1     1
+2     2
+3     3
+4     4
+5     5
+6     6
+7     7
+```
+
+## Sparklyr.flint
+
+[Sparklyr.flint](https://github.com/r-spark/sparklyr.flint) is a sparklyr extension that aims to make functionalities from the [Flint](https://github.com/twosigma/flint) time series library easily accessible from R. It is currently under active development. One good news is while the original [Flint](https://github.com/twosigma/flint) library was designed to work with Spark 2.x, a slightly modified [fork](https://github.com/yl790/flint) of it will work well with Spark 3.0, and within the existing sparklyr extension framework, sparklyr.flint can automatically determine which version of the Flint library to load based on the version of Spark it's connected to. Another good news is, as previously mentioned, sparklyr.flint doesn't know too much about its own destiny yet. Maybe you can play an active part in shaping its future!
+
+## EMR 6.0
+
+This release also features a small but important change that allows sparklyr to correctly connect to the version of Spark 2.4 that is included in Amazon EMR 6.0.
+
+Previously, sparklyr automatically assumes any Spark 2.x it is connecting to is built with Scala 2.11 and attempts to load any required Scala artifacts built with Scala 2.11 as well. This becomes problematic when connecting to Spark 2.4 from Amazon EMR 6.0, which is built with Scala 2.12. Starting from sparklyr 1.3, such problem can be fixed by simply specifying `scala_version = "2.12"` when calling `spark_connect` (e.g., `spark_connect(master = "yarn-client", scala_version = "2.12")`).
+
+## Spark 3.0
+
+Last but not least, it is worthwhile to mention sparklyr 1.3.0 is known to be fully compatible with the recently released Spark 3.0. We highly recommend upgrading your copy of sparklyr to 1.3.0 if you plan to have Spark 3.0 as part of your data workflow in future.
+
+## Acknowledgement
```

Comment:
Last time someone complained to me in Twitter that they were "left out" in the acknowledgements, I ended up fixing the blog post, but just passing this along. I think is awesome that we are doing this, but at times it can get tricky to not hurt someones feeling.  Not sure what is the best way to compute the acknowledgements here, but hopefully we are not forgetting someone.

Reply:
Wow... I shall double-check then
Another factor to consider is sometimes depending on when the release branch is cut their contribution may be considered the 1st (-ish) commit as part of the next release instead of the last (-ish) commit in the current release.

Reply:
^^ Oh yeah actually I did consider Jozef Hajnala's commit as the first one towards this release... I think I'd rather be err-ing on the "safe" side and just double-count contributors whose commits are too close to the branch cut :smile: